packages/relay/src/lib/eth.ts

```diff
@@ -1109,69 +1110,96 @@ export class EthImpl implements Eth {
 
-        await this.mirrorNodeClient.getContractRevertReasonFromTransaction(e, requestId, requestIdPrefix);
+    this.logger.error(e,
+        `${requestIdPrefix} Failed sendRawTransaction during record retrieval for transaction ${transaction}, returning computed hash`);
+    //Return computed hash if unable to retrieve EthereumHash from record due to error
+    return EthImpl.prepend0x(createHash('keccak256').update(transactionBuffer).digest('hex'));
+  }
 
-        this.logger.error(e,
-          `${requestIdPrefix} Failed sendRawTransaction during record retrieval for transaction ${transaction}, returning computed hash`);
-        //Return computed hash if unable to retrieve EthereumHash from record due to error
-        return EthImpl.prepend0x(createHash('keccak256').update(transactionBuffer).digest('hex'));
-      }
-    } catch (e: any) {
-      this.logger.error(e,
-        `${requestIdPrefix} Failed to successfully submit sendRawTransaction for transaction ${transaction}`);
-      if (e instanceof JsonRpcError) {
-        return e;
+  /**
+   * Submits a transaction to the network for execution.
+   *
+   * @param transaction
+   */
+  async sendRawTransaction(transaction: string, requestId?: string): Promise<string | JsonRpcError> {
+    const requestIdPrefix = formatRequestIdMessage(requestId);
+    if(transaction?.length >= constants.FUNCTION_SELECTOR_CHAR_LENGTH)
+      this.ethExecutionsCounter.labels(EthImpl.ethSendRawTransaction, transaction.substring(0, constants.FUNCTION_SELECTOR_CHAR_LENGTH)).inc();
+
+    const parsedTx = await this.parseRawTxAndPrecheck(transaction, requestId);
+    const transactionBuffer = Buffer.from(EthImpl.prune0x(transaction), 'hex');
+
+    let txSubmitted = false;
+    try {
+      const contractExecuteResponse = await this.hapiService.getSDKClient().submitEthereumTransaction(transactionBuffer, EthImpl.ethSendRawTransaction, requestId);
+      txSubmitted = true;
+      // Wait for the record from the execution.
+      let txId = contractExecuteResponse.transactionId.toString();
+      const formattedId = formatTransactionId(txId);
+      const  record = await this.mirrorNodeClient.repeatedRequest('getContractResult', [formattedId], 3, requestId);
+      if (!record) {
+        this.logger.warn(`${requestIdPrefix} No record retrieved`);
+        const tx = await this.mirrorNodeClient.getTransactionById(txId, 0, requestId);
+        if (tx.transactions?.length) {
+          const result = tx.transactions[0].result;
+          if (result === 'WRONG_NONCE') {
+            const accountInfo = await this.mirrorNodeClient.getAccount(parsedTx.from!, requestId)
+            const accountNonce = accountInfo.ethereum_nonce;
+            if (parsedTx.nonce > accountNonce) {
+              throw predefined.NONCE_TOO_HIGH(parsedTx.nonce, accountNonce);
+            }
+
+            throw predefined.NONCE_TOO_LOW(parsedTx.nonce, accountNonce);
+          }
+        }
+        throw predefined.INTERNAL_ERROR();
```

Comment:
nit: add a message in the error to note there was no matching transactions.

Reply:
✅  done